src/key_pair.rs

```diff
@@ -211,23 +216,25 @@ impl TryFrom<Vec<u8>> for KeyPair {
 
 impl KeyPair {
 	/// Generate a new random key pair for the specified signature algorithm
-	pub fn generate(alg: &'static SignatureAlgorithm) -> Result<Self, RcgenError> {
-		let system_random = SystemRandom::new();
+	pub fn generate(
+		alg: &'static SignatureAlgorithm,
+		rng: &dyn SecureRandom,
```

Comment:
This is still exposing ring as part of the public API.

Reply:
We could create our own `SecureRandom` trait, which is a copy of ring's. Then we could make a (private) struct wrapper of that, which then implements ring's `SecureRandom` trait. We would then take anything that impls our `SecureRandom` trait.

Reply:
I'll defer it to another PR.

Reply:
@thomaseizinger fine by me!